### PR TITLE
feat(pci-gateway): add datagrid, filter and pagination

### PR DESCRIPTION
**Pull request by kqesar**
| Question         | Answer
| ---------------- | ---
| Branch?          | `feat/DTCORE-2008`
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | DTCORE-2119
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [ ] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [ ] Breaking change is mentioned in relevant commits

## Description

Add datagrid, filters and pagination

### CODE REVIEW

File: packages/manager/apps/pci-gateway/src/api/data/gateway.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+) => {
+  return {
+    rows: items.slice(
+      pagination.pageIndex * pagination.pageSize,
+      (pagination.pageIndex + 1) * pagination.pageSize,
+    ),
+    pageCount: Math.ceil(items.length / pagination.pageSize),
+    totalRows: items.length,
+  };
+};
```
Comment (chipp972) [inline packages/manager/apps/pci-gateway/src/api/data/gateway.ts:30]:
```suggestion
) => ({
    rows: items.slice(
      pagination.pageIndex * pagination.pageSize,
      (pagination.pageIndex + 1) * pagination.pageSize,
    ),
    pageCount: Math.ceil(items.length / pagination.pageSize),
    totalRows: items.length,
  });
```

File: packages/manager/apps/pci-gateway/src/api/data/gateway.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  projectId: string,
+): Promise<GatewayResponse> => {
+  const { data } = await v6.get<GatewayResponse>(
+    `/cloud/project/${projectId}/aggregated/gateway`,
+  );
+  return data;
+};
+
+export const paginateResults = (
+  items: Gateway[],
```
Comment (chipp972) [inline packages/manager/apps/pci-gateway/src/api/data/gateway.ts:19]:
```suggestion
  items: unknown[],
```

It doesn't seem specific to Gateways
We could put this function next to the datagrid in the manager-components maybe

File: packages/manager/apps/pci-gateway/src/api/hooks/useGateway.ts [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+          ...row,
+          model: row.model.toUpperCase(),
+          connectedNetworkCount: uniqBy(row.interfaces, 'subnetId').length,
+          formattedIps:
+            row.externalInformation.ips.map((ip) => ip.ip).join(', ') || '',
+        };
+      });
+    },
+  });
+};
```
Comment (chipp972) [inline packages/manager/apps/pci-gateway/src/api/hooks/useGateway.ts:29]:
```suggestion
export const useAllAggregatedGateway = (projectId: string) => useQuery({
    queryKey: ['project', projectId, 'gateway'],
    queryFn: () => getAllAggregatedGateway(projectId),
    retry: false,
    select: (data): Gateway[] => data.resources.map((row) => ({
          ...row,
          model: row.model.toUpperCase(),
          connectedNetworkCount: uniqBy(row.interfaces, 'subnetId').length,
          formattedIps:
            row.externalInformation.ips.map((ip) => ip.ip).join(', ') || '',
        })),
  });
```

File: packages/manager/apps/pci-gateway/src/interface/index.ts [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+  response?: {
+    headers?: {
+      [key: string]: string;
+      'x-ovh-queryid': string;
+    };
+    data?: {
+      message?: string;
+    };
+  };
+}
```
Comment (chipp972) [inline packages/manager/apps/pci-gateway/src/interface/index.ts:17]: we export an APIError type from @ovh-ux/manager-core-api

**Review by sidlynx — APPROVED** [review]

**Review by seven-amid — APPROVED** [review]

**Pull request merged**